src/travel-details-screens/components/FlexibleTransportBookingDetails.tsx

```diff
@@ -0,0 +1,182 @@
+import {FlexibleTransportMessageBox} from './FlexibleTransportMessageBox';
+import {useNow} from '@atb/utils/use-now';
+import {useRemoteConfig} from '@atb/RemoteConfigContext';
+
+import {Dimensions} from 'react-native';
+import {useFirestoreConfiguration} from '@atb/configuration/FirestoreConfigurationContext';
+import {useSafeAreaInsets} from 'react-native-safe-area-context';
+
+const {width, height} = Dimensions.get('window');
+const isSmallScreen = width < 320 || height < 568;
+
+type FlexibleTransportBookingDetailsProps = {
+  leg: Leg;
+  close: () => void;
+};
+
+export const FlexibleTransportBookingDetails: React.FC<
+  FlexibleTransportBookingDetailsProps
+> = ({leg, close}) => {
+  const {t, language} = useTranslation();
+  const style = useStyle();
+  const {theme} = useTheme();
+
+  const {flex_booking_number_of_days_available} = useRemoteConfig();
+  const {configurableLinks} = useFirestoreConfiguration();
+
+  const publicCode = getPublicCodeFromLeg(leg);
+
+  const now = useNow(2500);
+  const bookingIsAvailable = getLegBookingIsAvailable(
+    leg,
+    now,
+    flex_booking_number_of_days_available,
+  );
+
+  const {bottom: safeAreaBottom} = useSafeAreaInsets();
+  const marginBottom =
+    safeAreaBottom > 0 ? safeAreaBottom : theme.spacings.medium;
+
+  return (
+    <BottomSheetContainer fullHeight={true} maxHeightValue={0.83}>
+      <ScreenHeaderWithoutNavigation
+        title={t(
+          TripDetailsTexts.flexibleTransport.needsBookingWhatIsThisTitle(
+            publicCode,
+          ),
+        )}
+        color="background_1"
+        setFocusOnLoad={false}
+        leftButton={{
+          type: 'close',
+          onPress: close,
+          text: t(ScreenHeaderTexts.headerButton.close.text),
+        }}
+      />
+
+      <View style={[style.scrollViewContainer, {marginBottom}]}>
+        <ScrollView contentContainerStyle={{padding: theme.spacings.xLarge}}>
+          <View style={style.messageBoxContainer}>
+            <FlexibleTransportMessageBox
+              leg={leg}
+              publicCode={publicCode}
+              now={now}
+              showStatusIcon={true}
+            />
+          </View>
+
+          {!isSmallScreen && (
+            <View style={style.imageContainer}>
+              <FlexibleTransport height={107} />
+            </View>
+          )}
+
+          <ThemeText type="heading__title" style={style.title}>
+            {t(TripDetailsTexts.flexibleTransport.contentTitle(publicCode))}
+          </ThemeText>
+          <View style={style.steps} accessibilityRole="list">
+            {
+              // eslint-disable-next-line rulesdir/translations-warning
+              TripDetailsTexts.flexibleTransport.steps.map((step, i) => {
+                const stepNumberText = `${i + 1}. `;
+                const stepInstructionText = t(step);
+                return (
+                  <View
+                    key={i}
```

Comment:
Shouldntt use index as key. Might not give unintended side effects here, but generally it is good to not use the anti-pattern. Can just use the text as key.